components/feature/app-links/src/test/java/mozilla/components/feature/app/links/SimpleRedirectDialogFragmentTest.kt

```diff
@@ -0,0 +1,98 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.app.links
+
+import android.widget.Button
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentTransaction
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import mozilla.components.support.test.mock
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.doReturn
+import org.mockito.Mockito.spy
+
+@RunWith(AndroidJUnit4::class)
+class SimpleRedirectDialogFragmentTest {
+    private val webUrl = "https://example.com"
+
+    @Test
+    fun `Dialog confirmed callback is called correctly`() {
+        var onConfirmCalled = false
+        var onCancelCalled = false
+
+        val onConfirm = { onConfirmCalled = true }
+        val onCancel = { onCancelCalled = true }
+        val fragment = spy(SimpleRedirectDialogFragment.newInstance(themeResId = R.style.Theme_AppCompat_Light))
+        doReturn(testContext).`when`(fragment).requireContext()
+        doReturn(mockFragmentManager()).`when`(fragment).fragmentManager
+        fragment.setAppLinkRedirectUrl(webUrl)
+        fragment.onConfirmRedirect = onConfirm
+        fragment.onCancelRedirect = onCancel
+
+        val dialog = fragment.onCreateDialog(null)
+        dialog.show()
+
+        val confirmButton = dialog.findViewById<Button>(android.R.id.button1)
+        confirmButton?.performClick()
+        assertTrue(onConfirmCalled)
+        assertFalse(onCancelCalled)
+    }
+
+    @Test
+    fun `Dialog cancel callback is called correctly`() {
+        var onConfirmCalled = false
+        var onCancelCalled = false
+
+        val onConfirm = { onConfirmCalled = true }
+        val onCancel = { onCancelCalled = true }
+        val fragment = spy(SimpleRedirectDialogFragment.newInstance(themeResId = R.style.Theme_AppCompat_Light))
+        doReturn(testContext).`when`(fragment).requireContext()
+        doReturn(mockFragmentManager()).`when`(fragment).fragmentManager
+        fragment.setAppLinkRedirectUrl(webUrl)
+        fragment.onConfirmRedirect = onConfirm
+        fragment.onCancelRedirect = onCancel
+
+        val dialog = fragment.onCreateDialog(null)
+        dialog.show()
+
+        val confirmButton = dialog.findViewById<Button>(android.R.id.button2)
+        confirmButton?.performClick()
+        assertFalse(onConfirmCalled)
+        assertTrue(onCancelCalled)
+    }
+
+    @Test
+    fun `Dialog confirm and cancel is not called when dismissed`() {
```

Comment:
Just curious, what is the behaviour here if the user dismisses? We won't be launching the app, and also won't be loading the URL, right? So essentially, we do nothing in that case?

Reply:
Yes, though we `setCancelable` to false, but the user can still back out or tap outside the dialog to dismiss it.  Do you want us to load the URL if the user dismisses the dialog without tapping on the buttons?

Reply:
Figured out that the fragment should be set to false in `setCancelable`.  This prevents user tapping on the outside of the dialog to dismiss it.  In practice the user shouldn't be able to dismiss the dialog anymore.  I'll keep the test though since its on the dialog level.

Reply:
No, I am fine :). Was just curious about it.